### PR TITLE
WIP: Add functionality test for mentors automation

### PR DESCRIPTION
**Pull request by nverbic**
## Description
 
Addressing review suggestions to create simple test to check write and append is working as expected:
https://github.com/WomenCodingCommunity/WomenCodingCommunity.github.io/pull/214

This is a draft for the first (functionality) test for the write mode - create new temp yml file with all mentors from the xlsx table.

Also using some files from @dricazenck draft PR https://github.com/WomenCodingCommunity/WomenCodingCommunity.github.io/pull/180

Testing basic functionality

https://github.com/WomenCodingCommunity/WomenCodingCommunity.github.io/issues/130

## Change Type
- [ ] Bug Fix
- [ ] New Feature
- [ ] Code Refactor
- [ ] Mentor Update
- [ ] Documentation
- [x] Other

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] I checked and followed the [contributor guide](CONTRIBUTING.md) 
- [x] I have tested my changes locally.
- [ ] I have added a screenshot from the website after I tested it locally

### CODE REVIEW

File: tools/tests/create_new_yml_test.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Given XLSX_1 When no mentors Then create new Yml
+# Given XLSX_1 When existent yml Then Merge Yml's
+# Given XLSX_2 When existent yml Then Merge Yml
+# Given XLSX_2 When All mentors are present Then no change happen
```
Comment (dricazenck) [inline tools/tests/create_new_yml_test.py]: This you can remove.

File: tools/tests/create_new_yml_test.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import pytest
+from file_utils import TOOLS_PATH
+from automation import run_automation, read_yml_file, WriteMode
+
+MENTOR_ADRIANA = "Adriana Zencke Zimmermann"
+MENTOR_A = "Mentor A"
+MENTOR_B = "Mentor B"
+MENTOR_C = "Mentor C"
+
+def test_write_mentors_skip_zero_rows(monkeypatch):
```
Comment (dricazenck) [inline tools/tests/create_new_yml_test.py]: You don't need to create a single file for unit test, you can create one test file with the same name as original source, but with _test only it is too big you need to split, or per domain as we discussed.

File: tools/tests/create_new_yml_test.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    run_automation()
+
+    result = read_yml_file(tmp_filename)
+    assert len(result) == 4, f"Expected to write 4 mentors but added {len(result)}"
+    assert MENTOR_ADRIANA == result[0]['name'], f"Expected content to be {MENTOR_ADRIANA} but got '{result[0]['name']}'"
+    assert MENTOR_A == result[1]['name'], f"Expected content to be {MENTOR_A} but got '{result[1]['name']}'"
+    assert MENTOR_B == result[2]['name'], f"Expected content to be {MENTOR_B} but got '{result[2]['name']}'"
+    assert MENTOR_C == result[3]['name'], f"Expected content to be {MENTOR_C} but got '{result[3]['name']}'"
+
+    # Clean up the temporary file
```
Comment (dricazenck) [inline tools/tests/create_new_yml_test.py]: The code is clear 🥰 no comment

File: tools/tests/create_new_yml_test.py (hunk 1 above)
Comment (nverbic) [reply tools/tests/create_new_yml_test.py]: Yes :-)

**Review by dricazenck — APPROVED** [review]

**Pull request merged**